ocaml/libs/stunnel/stunnel.ml

```diff
@@ -157,6 +157,9 @@ let pool =
   ; cert_bundle_path= "/etc/stunnel/xapi-pool-ca-bundle.pem"
   }
 
+let external_host ext_host_cert_file =
+  {sni= None; verify= VerifyPeer; cert_bundle_path= ext_host_cert_file}
```

Comment:
Should this be `CheckHost`?

Reply:
Per the stunnel spec:
```
verifyPeer = yes | no
verify the end-entity (leaf) peer certificate

The end-entity (leaf) peer certificate needs to be stored either in the file specified with CAfile, or in the directory specified with CApath.

default: no

```
and

```
checkHost = HOST
verify the host of the end-entity (leaf) peer certificate subject

Certificates are accepted if no subject checks were specified, or the host name of the end-entity (leaf) peer certificate matches any of the hosts specified with checkHost.

Multiple checkHost options are allowed in a single service section.

This option requires OpenSSL 1.0.2 or later.
```

I think it is VerifyPeer, as we only need to verify the leaf peer certificate, no subject or host name verification is required.

Reply:
> we only need to verify the leaf peer certificate, no subject or host name verification is required.
What would happen if a root CA certificate is passed? How can such a CA certificate verify the leaf without going through the chain?

Reply:
I think the certificate required here is the host certificate of the external host, get via `host.get_server_certificate`.